sklearn/compose/_column_transformer.py

```diff
@@ -445,7 +466,10 @@ def fit_transform(self, X, y=None):
         self._validate_column_callables(X)
         self._validate_remainder(X)
 
-        result = self._fit_transform(X, y, _fit_transform_one)
+        fit_transform_one = partial(
+            _fit_transform_one, is_transform=True, clsname='ColumnTransformer')
```

Comment:
It is possible to integrate your suggestion and move some logic into `ColumnTransformer._fit_transform. I think this would make the code a little more maintainable.